sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextBase.java

```diff
@@ -131,11 +131,23 @@ public CosmosQueryRequestOptions getFeedOptions(String continuationToken, Intege
         Map<String, String> requestHeaders = new HashMap<>();
 
         ConsistencyLevel defaultConsistencyLevel = this.client.getDefaultConsistencyLevelAsync();
-        ConsistencyLevel desiredConsistencyLevel = this.client.getDesiredConsistencyLevelAsync();
+        ConsistencyLevel desiredConsistencyLevel = cosmosQueryRequestOptions.getConsistencyLevel() != null ?
+            cosmosQueryRequestOptions.getConsistencyLevel():
+            this.client.getDesiredConsistencyLevelAsync();
+
+        boolean sessionTokenApplicable =
```

Comment:
What are session-container semantics with request level overrides? 

ex: With eventual and session SxS
- Is session expected to progress even with eventual requests context?

Reply:
It is independent of the change here. Backend ignores the SessionToken when we request Eventual Consistency on a request level. The response still contains a session token - even when request is for eventual without session token.  We update the SessionContainer with the session token from the response (if it is newer). So session would progress even for individual eventual consistency requests.